### ISSUE TITLE
How to change keyboard shortcuts?

### ISSUE THREAD

**Issue by TheYellowArchitect** [LukeSmithxyz/voidrice#993]
Configuration file is at /home/username/.local/src/dwm/config.h
I change (around line 237) `SHMCD("tutorialvids")` to `SHMCD("youtubedl-clipboard")`

As for this script, I made it and put it onto /home/username/.local/bin, and it has the same chmod as `tutorialvids`
Don't bother debugging it, it certainly works, but let me offer you the code because asking for help, the more information the better.
```
#!/bin/sh

# If the windows drive is connected (aka have access to youtubedlvideos)
if [ -d "/mnt/YoutubeDLVideos" ];
then
	targetFile="/mnt/YoutubeDLVideos"
else # place the video in an existing folder (home folder)
	targetFile="/home/username"
fi

# take the clipboard's selection, and give it as argument to yt-dlp
xclip -o | xargs -r yt-dlp -P $targetFile
```

I save the config.h with the above, and type (while in that directory) `sudo make clean install`, yet nothing happens. Like, tutorialvids is still mapped to Super+F2, until I reboot, and when I reboot, Super+F2 does nothing at all.

I want to map some other minor things (opening file locations instantly with `lf`, so I don't browse by hand all day), but I just can't map them. I don't know why, `sudo make install` used to work, I remapped something in the past :(

**Comment by TheYellowArchitect** [comment]
I tried going in `~/.xprofile`, and on line 13, added "F2 &", and ofc `:wq`
Why I did this? Because idk how the scripts are enabled and activated by dwm, so I guessed this is it. It crashed my desktop to black screen, had to login via tty2 to undo the above change.

The contents of the F2 script is 
```
#!/bin/sh

lf /mnt/Users/unity/Desktop/Misc/kk
```literally 1 command which certainly works on the terminal. I just want to replace the F2 of tutorial vids with the above "F2" script in dwm's config.h

Has anyone made his own .local/bin scripts and placed them in dwm/config.h and it all just worked? (Placing the code logic in config.h works, but not the script file)

**Comment by kronikpillow** [comment]
Wow your complicating something so simpče xD F2 & to .xprofile?? LMAO xD dude, just place the scripts in .local/bin or any old or new subdirectory in .local/bin ... Bind the script to a keybinding, recompile and install dwm again after rebinding the keys, then hit mod+backspace and refresh dwm to load the new bindings

**Comment by TheYellowArchitect** [comment]
It is true, I am kinda lost on what I'm doing wrong, and when this failed, I came here to ask for help

>just place the scripts in .local/bin or any old or new subdirectory in .local/bin
This was done

>Bind the script to a keybinding
How?

This is /dwm/config.h, line 233
```
{ MODKEY,			XK_Insert,	spawn,		SHCMD("xdotool type $(cat ~/.local/share/larbs/snippets | dmenu -i -l 50 | cut -d' ' -f1)") },

	{ MODKEY,			XK_F1,		spawn,		SHCMD("groff -mom /usr/local/share/dwm/larbs.mom -Tpdf | zathura -") },
	{ MODKEY,			XK_F2,		spawn,		SHCMD("youtubedl-clipboard") },
	{ MODKEY,			XK_F3,		spawn,		SHCMD("") },
	{ MODKEY,			XK_F4,		spawn,		SHCMD("") },
	/* { MODKEY,			XK_F5,		xrdb,		{.v = NULL } }, */
	{ MODKEY,			XK_F6,		spawn,		SHCMD("torwrap") },
	{ MODKEY,			XK_F7,		spawn,		SHCMD("td-toggle") },
	{ MODKEY,			XK_F8,		spawn,		SHCMD("mw -Y") },
	{ MODKEY,			XK_F9,		spawn,		SHCMD("dmenumount") },
	{ MODKEY,			XK_F10,		spawn,		SHCMD("dmenuumount") },
```

Note F2. No matter what script I put inside "", it doesn't work, even a simple echo

>recompile and install dwm again after rebinding the keys
`sudo make clean install` while on the file location of config.h, correct?

**Comment by TheYellowArchitect** [comment]
@LukeSmithxyz I understand that you don't like documentation or answering basic questions, and also that my post is kinda bloated.

I went to Linux because I saw your setup, and indeed it is superior to windows I was used to, and being able to bind scripts on keys is literally the final step for me to make the final jump onto it without any regrets.
Could you re-open the issue so at least someone else answers, or answer briefly how to put a script on key mapping? 
I ask this because currently I cannot configure my own system, which is the main point of this setup. Maybe I have a bug? Doubtful, probably retardation, so I may have missed a step somewhere or something.
I also cannot ask anywhere else, since this is the only place which has so similar a setup to mine (pretty much identical)

**Comment by decantr** [comment]
Couple of basic things to check:

Is your script marked as executable? `chmod +x /path/to/script`
Is your `config.h` actually in the dwm source folder?
If you are using `echo` to test your script is running, how are you capturing that? Might be better to write to a temporary file in the script so you can see it working?

**Comment by TheYellowArchitect** [comment]
> Is your script marked as executable? `chmod +x /path/to/script`
> Is your `config.h` actually in the dwm source folder?

Yup to both

> If you are using `echo` to test your script is running, how are you capturing that? Might be better to write to a temporary file in the script so you can see it working?

This is the script pretty much, and typing it onto the terminal, outputs test
```
#!/bin/sh

echo "test"
```

I could replace that final line with `lf`, which ofc works on the terminal by typing the script name (opens lf in a seperate window)
But it doesn't run by pressing Super+F2.

As for recompiling, I just run `sudo make install` while inside the folder dwm (which has config.h inside)

Note 1: Putting "lf" in the config.h's rebinding works, but a script-name which contains "lf" doesn't work. All aforementioned scripts are in the same folder as luke's scripts, which is .local/bin

Note 2: Recompiling dwm by itself has no issues, as I can put torbrowser-launcher into some other binding, its just the user scripts which can't be binded even after recompiling.

@decantr @kronikpillow

**Comment by TheYellowArchitect** [comment]
Looking back at this, I found the solution and I share it in hopes whoever wants to do something similar won't waste time or bloat his config

`lf` **runs exclusively by terminal.**
The above F2 script runs normally on terminal and has proper results/output.
But on dmenu, it did not. And conclusively, it will not on dmenu either.

To fix this, I looked into how luke has implemented Super+R in dwm/config.h, and it turns out he has `SHCMD(TERMINAL "-e lf")`
So if you want to map a button to go to any location (or open lf and do anything), pretty much copy-paste it and put your location after, e.g.
`{ MODKEY,				XK_F2,		spawn,		SHCMD(TERMINAL " -e lf /harddrive/Users/username/pdf") },`

Hope this helps :)
